src/backend/access/transam/varsup.c

```diff
@@ -24,12 +26,16 @@
 #include "postmaster/autovacuum.h"
 #include "storage/pmsignal.h"
 #include "storage/proc.h"
+#include "utils/guc.h"
 #include "utils/syscache.h"
 
 
 /* Number of OIDs to prefetch (preallocate) per XLOG write */
 #define VAR_OID_PREFETCH		8192
 
+#define OID_TO_BUFFER_START(oid) 		((oid) + INT_MIN)
```

Comment:
why do we need to do +/- INT_MIN? we can keep all involved vars as uint32 for simplicity

Reply:
The value temp_oid_buffer_start is an int, which is why the Oids needs to be properly converted. But we must treat it like a uint32 by adjusting it because it might store an oid buffer start value greater than INT_MAX.

Reply:
We can keep all variables as unit32. If GUC is int, we just convert it to uint32 when accessing its value. That way tempoid logic is not polluted with INT->UINT conversion.

Reply:
I will hold further review till we do this change as it would simply logic and review